v2/tools/generator/internal/armconversion/convert_to_arm_function_builder.go

```diff
@@ -537,13 +537,24 @@ func (builder *convertToARMBuilder) buildToPropInitializer(
 	}
 }
 
-func (builder *convertToARMBuilder) propertiesWithSameNameHandler(
+func (builder *convertToARMBuilder) propertiesByNameHandler(
 	toProp *astmodel.PropertyDefinition,
 	fromType *astmodel.ObjectType,
 ) (propertyConversionHandlerResult, error) {
 
-	fromProp, ok := fromType.Property(toProp.PropertyName())
-	if !ok {
+	fromProp, found := fromType.Property(toProp.PropertyName())
+	if !found {
+		// No direct match by name, look for renames
+		// Look for a candidate fromProp whose original name matches toProp
+		fromType.Properties().ForEach(func(p *astmodel.PropertyDefinition) {
```

Comment:
This came out pretty clean. 

You also could have done this w/ property tags as well if you wanted. How you did it works for me though.